### ISSUE TITLE
Automate installation with pyproject.toml dependency management

### ISSUE THREAD

**Issue by niznik-dev** [niznik-dev/predicting-zygosity#198]
## Summary

Automate the installation process using `pyproject.toml` so users can set up the environment with a single conda/pip command instead of manually running 7+ installation steps.

## Current State

Users must manually install dependencies in this order:
1. Create conda environment with Python 3.13
2. Install PyTorch with CUDA 12.6 from custom index
3. Install 10+ core packages via pip
4. Install torchtune (preferably nightly build)
5. Install cruijff_kit in editable mode
6. (Contributors only) Install gh CLI via conda

See README.md lines 42-142 for full manual process.

## Proposed Solution

Configure `pyproject.toml` with:

### Base Installation
- Define all core dependencies with proper version constraints
- Specify PyTorch CUDA 12.6 installation via pip `--find-links` or `--index-url`
- **Strongly recommend torchtune nightly build** (includes validation loss tracking)
- Support editable install: `pip install -e .`

### Optional Dependencies Groups
```toml
[project.optional-dependencies]
dev = [
    "pytest",
    "pytest-cov",
    # gh CLI must be installed via conda separately
]
```

Users would install with:
```bash
# Standard installation
conda create -n cruijff python=3.13 -y
conda activate cruijff
pip install -e .

# Dev installation (includes testing tools)
conda create -n cruijff python=3.13 -y
conda activate cruijff
pip install -e ".[dev]"
conda install -c conda-forge gh -y  # gh must be via conda
```

## Technical Considerations

1. **PyTorch CUDA 12.6**: Need to specify custom index URL in pyproject.toml
   - Use `--find-links` or dependency_links
   - Alternative: Document as a pre-installation step

2. **Torchtune Nightly**: Recommended for `val_loss` tracking
   - Consider making nightly the default with stable as fallback option
   - Could use optional dependency groups: `[project.optional-dependencies] stable = ["torchtune==0.6.1"]`

3. **GitHub CLI (gh)**: Must be installed via conda, not pip
   - Keep as separate conda install step for dev environment
   - Document clearly in installation section

4. **Conda vs Pip**: Some packages work better via conda
   - Evaluate if any current pip packages should move to conda

## Dependencies to Include

Core (all users):
- torch (CUDA 12.6)
- torchao
- wandb
- h5py
- inspect-ai
- inspect-viz
- datasets
- peft
- transformers
- scikit-learn
- matplotlib
- torchtune (nightly preferred)

Dev (contributors only):
- pytest
- pytest-cov
- gh (via conda, not pip - document separately)

## Testing Requirements

- [ ] Test fresh installation on HPC cluster
- [ ] Verify CUDA availability after PyTorch install
- [ ] Confirm torchtune nightly installs correctly
- [ ] Test both base and dev installations
- [ ] Verify editable mode works for local development

## Documentation Updates

- [ ] Update README.md installation section
- [ ] Add troubleshooting for common issues
- [ ] Document how to switch between torchtune stable/nightly
- [ ] Update claude.local.md.template if needed

## References

- Current installation: README.md lines 42-142
- Current pyproject.toml: Lines 1-16 (minimal configuration)
- Related: #203 (unit tests for setup_finetune.py - will use pytest)

**Comment by msalganik** [comment]
@niznik-dev this sounds great.  I ended up with something similar in a separate repo where I'm working on a torchtune-config-writer. Feel free to check out if anything here is helpful or something you want to avoid: https://github.com/msalganik/torchtune_config_writer/tree/master

I decided to try uv.

Also, please ignore the rest of the repo.  I'm exploring what a spec would look like to take one part that is currently done by Claude Code and replace it with Python code (which is much easier to test and force to handle edges cases in exactly the way we want).  It is still a work in progress.

**Comment by niznik-dev** [comment]
Interesting! I would imagine any things we all converge on are a good sign (adding @sarahepedersen). One thing I specifically did *not* do yet was set any standard for what the experiment design document should look like - it seemed likely that you both probably have more ideas and stronger opinions on that.
